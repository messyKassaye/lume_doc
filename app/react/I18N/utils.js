import * as Cookie from 'tiny-cookie';
import { isClient } from 'app/utils';

const languageInLanguages = (languages, locale) => Boolean(languages.find(l => l.key === locale));
const getURLLocale = (locale, languages = []) =>
  languageInLanguages(languages, locale) ? locale : null;
const getCookieLocale = (cookie, languages) =>
  cookie.locale && languageInLanguages(languages, cookie.locale) ? cookie.locale : null;
const getDefaultLocale = languages => (languages.find(language => language.default) || {}).key;

const I18NUtils = {
  getLocale(urlLanguage, languages, cookie = {}) {
    return (
      getURLLocale(urlLanguage, languages) ||
      getCookieLocale(cookie, languages) ||
      getDefaultLocale(languages)
    );
  },

  saveLocale: locale => {
    if (isClient) {
      Cookie.set('locale', locale, { expires: 365 * 10 });
    }
  },
};

export default I18NUtils;
