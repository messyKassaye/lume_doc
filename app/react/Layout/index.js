export { default as Item } from './Item';
export { default as TemplateLabel } from './TemplateLabel';
export { default as Icon } from './Icon';
export { default as Tip } from './Tip';
export { default as SidePanel } from './SidePanel';
export { default as DocumentLanguage } from './DocumentLanguage';
export { default as Thumbnail } from './Thumbnail';
export { default as ConfirmButton } from './ConfirmButton';
export { default as CurrentLocationLink } from './CurrentLocationLink';
export { default as Warning } from './Warning';
export { default as BackButton } from './BackButton';
