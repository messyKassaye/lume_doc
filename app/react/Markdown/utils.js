export const objectPath = (path, object) =>
  path.split('.').reduce((o, key) => {
    if (!o || !key) {
      return o;
    }
    return o.toJS ? o.get(key) : o[key];
  }, object);
