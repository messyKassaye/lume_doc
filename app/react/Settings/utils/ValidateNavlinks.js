export default function(links) {
  const validator = {
    '': {},
  };

  links.forEach((_link, index) => {
    validator[''][`links.${index}.title.required`] = form => {
      if (!form.links[index]) {
        return true;
      }
      const { title } = form.links[index];
      return Boolean(title && String(title).trim().length);
    };
  });

  return validator;
}
