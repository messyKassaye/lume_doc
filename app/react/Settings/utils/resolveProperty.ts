/** @format */
import { TemplateSchema } from 'shared/types/templateType';
import { ThesaurusSchema } from 'shared/types/thesaurusType';

/**
 * A given template property may refer to an existing thesaurus to provide
 * multi-select values. This function resolves a template property name that
 * refers to a particular thesaurus.
 */
export function resolveTemplateProp(thesaurus: ThesaurusSchema, templates: TemplateSchema[]) {
  let matchingProp;
  for (let i = 0; i < templates.length; i += 1) {
    const template = templates[i];
    const matchProp = template.properties?.find((prop: any) => prop.content === thesaurus._id);
    if (matchProp !== undefined) {
      matchingProp = matchProp;
      // TODO: Consider supporting multiple fields referring to the same thesaurus.
      break;
    }
  }
  return matchingProp;
}
