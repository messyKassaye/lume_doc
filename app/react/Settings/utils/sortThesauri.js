export default function sortThesauri(list) {
  list.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
  return list;
}
