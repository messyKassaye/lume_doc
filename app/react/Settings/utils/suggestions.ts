import { PropertySchema } from 'shared/types/commonTypes';

function baseQuery(templateID: string, includeUnpublished: boolean, unpublishedOnly: boolean) {
  const query = {
    select: ['sharedId'],
    limit: 1,
    filters: {},
    includeUnpublished,
    unpublished: unpublishedOnly,
    allAggregations: true,
    types: [templateID],
  };
  return query;
}

export function getReadyToReviewSuggestionsQuery(
  templateID: string,
  templateProperty: PropertySchema
) {
  if (!templateProperty || !templateProperty.name) {
    return null;
  }
  const { name } = templateProperty;
  return {
    ...baseQuery(templateID, true, false),
    filters: {
      [name]: {
        values: ['missing'],
      },
      [`__${name}`]: {
        values: ['any'],
      },
    },
  };
}

export function getReadyToPublishSuggestionsQuery(
  templateID: string,
  templateProperty: PropertySchema
) {
  if (!templateProperty || !templateProperty.name) {
    return null;
  }
  const { name } = templateProperty;
  return {
    ...baseQuery(templateID, false, true),
    filters: {
      [name]: {
        values: ['any'],
      },
      [`__${name}`]: {
        values: ['any'],
      },
    },
  };
}

export function getLabelsQuery(templateID: string, templateProperty: PropertySchema) {
  if (!templateProperty || !templateProperty.name) {
    return null;
  }
  const { name } = templateProperty;
  return {
    ...baseQuery(templateID, true, false),
    filters: {
      [name]: {
        values: ['any'],
      },
    },
  };
}
