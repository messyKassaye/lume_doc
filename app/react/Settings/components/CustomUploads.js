import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Dropzone from 'react-dropzone';
import Immutable from 'immutable';
import PropTypes from 'prop-types';
import React from 'react';

import { Thumbnail, ConfirmButton } from 'app/Layout';
import { actions } from 'app/BasicReducer';
import { t } from 'app/I18N';
import RouteHandler from 'app/App/RouteHandler';
import api from 'app/utils/api';
import { Icon } from 'UI';

import { uploadCustom, deleteCustomUpload } from '../../Uploads/actions/uploadsActions';

export class CustomUploads extends RouteHandler {
  static async requestState(requestParams) {
    const customUploads = await api.get('files', requestParams.add({ type: 'custom' }));
    return [actions.set('customUploads', customUploads.json)];
  }

  constructor(props, context) {
    super(props, context);
    this.onDrop = this.onDrop.bind(this);
  }

  onDrop(files) {
    files.forEach(file => {
      this.props.upload(file);
    });
  }

  render() {
    return (
      <div className="panel panel-default">
        <div className="panel-heading">{t('System', 'Custom Uploads')}</div>
        <div className="panel-body custom-uploads">
          <Dropzone className="upload-box" onDrop={this.onDrop}>
            <div className="upload-box_wrapper">
              <Icon icon="upload" />
              <button className="upload-box_link" type="button">
                Browse files to upload
              </button>
              <span> or drop your files here.</span>
            </div>
            {this.props.progress && (
              <div className="uploading">
                <Icon icon="spinner" spin /> Uploading ...
              </div>
            )}
          </Dropzone>
          <ul>
            {this.props.customUploads.map(upload => (
              <li key={upload.get('filename')}>
                <Thumbnail file={`/assets/${upload.get('filename')}`} />
                <div className="info">
                  URL:
                  <br />
                  <span className="thumbnail-url">{`/assets/${upload.get('filename')}`}</span>
                  <ConfirmButton action={() => this.props.deleteCustomUpload(upload.get('_id'))}>
                    Delete
                  </ConfirmButton>
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

CustomUploads.defaultProps = {
  progress: false,
};

CustomUploads.propTypes = {
  progress: PropTypes.bool,
  customUploads: PropTypes.instanceOf(Immutable.List).isRequired,
  upload: PropTypes.func.isRequired,
  deleteCustomUpload: PropTypes.func.isRequired,
};

export const mapStateToProps = ({ customUploads, progress }) => ({
  customUploads,
  progress: !!progress.filter((_v, key) => key.match(/customUpload/)).size,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ upload: uploadCustom, deleteCustomUpload }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CustomUploads);
