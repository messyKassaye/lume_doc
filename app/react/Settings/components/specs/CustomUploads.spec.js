import Immutable from 'immutable';
import React from 'react';

import { shallow } from 'enzyme';
import api from 'app/utils/api';
import { ConfirmButton } from 'app/Layout';
import { RequestParams } from 'app/utils/RequestParams';

import { CustomUploads, mapStateToProps } from '../CustomUploads';

describe('CustomUploads', () => {
  let component;
  let props;
  let context;

  beforeEach(() => {
    spyOn(api, 'get').and.returnValue(Promise.resolve({ json: 'uploads' }));
    props = {
      upload: jasmine.createSpy('upload'),
      deleteCustomUpload: jasmine.createSpy('deleteCustomUpload'),
      customUploads: Immutable.fromJS([]),
    };
  });

  const render = () => {
    context = { store: { getState: () => ({}), dispatch: jasmine.createSpy('dispatch') } };
    component = shallow(<CustomUploads {...props} />, { context });
  };

  it('should render CustomUploads component with uploaded files', () => {
    props.customUploads = Immutable.fromJS([{ filename: 'file1' }, { filename: 'file2' }]);
    render();
    expect(component).toMatchSnapshot();
  });

  describe('when upload on progress', () => {
    it('should render on progress feedback', () => {
      props.progress = true;
      render();
      expect(component).toMatchSnapshot();
    });
  });

  describe('deleteCustomUpload', () => {
    it('should call deleteCustomUpload on click', () => {
      props.customUploads = Immutable.fromJS([{ _id: 'upload', filename: 'name' }]);
      render();

      component
        .find(ConfirmButton)
        .props()
        .action();

      expect(props.deleteCustomUpload).toHaveBeenCalledWith('upload');
    });
  });

  describe('mapStateToProps', () => {
    it('should map current progress and files to props', () => {
      const state = {
        customUploads: 'customUploads',
        progress: Immutable.fromJS({}),
      };

      props = mapStateToProps(state);
      expect(props.customUploads).toBe('customUploads');
      expect(props.progress).toBe(false);

      state.progress = Immutable.fromJS({
        customUpload_unique_id: 1,
        customUpload_unique_id2: 100,
      });
      props = mapStateToProps(state);
      expect(props.progress).toBe(true);

      state.progress = Immutable.fromJS({ not_custom_upload: 9 });
      props = mapStateToProps(state);
      expect(props.progress).toBe(false);
    });
  });

  describe('requestState', () => {
    it('should get the uploads', async () => {
      const request = new RequestParams();
      const actions = await CustomUploads.requestState(request);

      const expectedParams = new RequestParams({ type: 'custom' });
      expect(api.get).toHaveBeenCalledWith('files', expectedParams);
      expect(actions).toMatchSnapshot();
    });
  });
});
