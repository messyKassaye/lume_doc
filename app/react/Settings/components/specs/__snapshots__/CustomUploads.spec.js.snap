// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomUploads requestState should get the uploads 1`] = `
Array [
  Object {
    "type": "customUploads/SET",
    "value": "uploads",
  },
]
`;

exports[`CustomUploads should render CustomUploads component with uploaded files 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Custom Uploads
  </div>
  <div
    className="panel-body custom-uploads"
  >
    <t
      className="upload-box"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDrop={[Function]}
      preventDropOnDocument={true}
    >
      <div
        className="upload-box_wrapper"
      >
        <Connect(Icon)
          icon="upload"
        />
        <button
          className="upload-box_link"
          type="button"
        >
          Browse files to upload
        </button>
        <span>
           or drop your files here.
        </span>
      </div>
    </t>
    <ul>
      <li
        key="file1"
      >
        <Thumbnail
          alt=""
          file="/assets/file1"
        />
        <div
          className="info"
        >
          URL:
          <br />
          <span
            className="thumbnail-url"
          >
            /assets/file1
          </span>
          <ConfirmButton
            action={[Function]}
            className=""
            message="Are you sure you want to continue?"
            title="Confirm action"
          >
            Delete
          </ConfirmButton>
        </div>
      </li>
      <li
        key="file2"
      >
        <Thumbnail
          alt=""
          file="/assets/file2"
        />
        <div
          className="info"
        >
          URL:
          <br />
          <span
            className="thumbnail-url"
          >
            /assets/file2
          </span>
          <ConfirmButton
            action={[Function]}
            className=""
            message="Are you sure you want to continue?"
            title="Confirm action"
          >
            Delete
          </ConfirmButton>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`CustomUploads when upload on progress should render on progress feedback 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    Custom Uploads
  </div>
  <div
    className="panel-body custom-uploads"
  >
    <t
      className="upload-box"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={true}
      onDrop={[Function]}
      preventDropOnDocument={true}
    >
      <div
        className="upload-box_wrapper"
      >
        <Connect(Icon)
          icon="upload"
        />
        <button
          className="upload-box_link"
          type="button"
        >
          Browse files to upload
        </button>
        <span>
           or drop your files here.
        </span>
      </div>
      <div
        className="uploading"
      >
        <Connect(Icon)
          icon="spinner"
          spin={true}
        />
         Uploading ...
      </div>
    </t>
    <ul />
  </div>
</div>
`;
